### ISSUE TITLE
CACHE_MANAGER BSOD

### ISSUE THREAD

**Issue by DavidRockin** [maharmstone/btrfs#504]
I've experienced a few `CACHE_MANAGER` blue screens recently, and they usually happen when some file operation is happening on my BTRFS partition (on an SSD, partition was creating in Linux), ex: manually deleting/creating files, or when .NET or Unity is doing something relatively I/O intensive (indexing, building a project)

**System Specs**
```
Windows 10 Pro N - 21H2 Build: 19044.1826
Latest BTRFS Driver 1.8
Pagefile size 1gb

i5 6600k 3.5ghz (not overclocked because my motherboard doesn't support it)
48GB DDR4

Btrfs partition of 530GB, on a 1TB SSD -- should be SATA_0

Chipset:
Intel 100 Series/C230 - A131
```

I've attached the 4 minidumps in the zip file and hope they're helpful.

Thanks

[BTRFS Dumps.zip](https://github.com/maharmstone/btrfs/files/9214027/BTRFS.Dumps.zip)

**edit**

I'll install the debug version, and hopefully when it happens again it might provide more information.

**Comment by maharmstone** [comment]
Thanks. Is this a preview version of Windows? These appear to be caused by assertions in CcCopyReadEx, which I thought would have been turned off in the Release version.

**Comment by DavidRockin** [comment]
Nope, it is not a preview build but the normal publicly released versions.

**Comment by maharmstone** [comment]
The assertions here was apparently `ValidDataLength <= FileSize <= AllocationSize`. I'm closing this as there doesn't seem to be anyway to trigger it with the current version of the driver. Please re-open this if it happens again.
